js_modules/dagit/packages/core/src/ticks/DynamicPartitionRequests.tsx

```diff
@@ -0,0 +1,72 @@
+import {Box, Colors, Icon, Subheading, Table, Tag} from '@dagster-io/ui';
+import React from 'react';
+
+import {DynamicPartitionsRequestType} from '../graphql/types';
+
+import {DynamicPartitionRequestFragment} from './types/SensorDryRunDialog.types';
+
+export function DynamicPartitionRequests({
+  requests,
+}: {
+  requests: DynamicPartitionRequestFragment[];
+}) {
+  const rows = React.useMemo(() => {
+    if (!requests.length) {
+      return [];
+    }
+    const rows: {key: string; def: string; type: DynamicPartitionsRequestType}[] = [];
+    requests.forEach(({partitionKeys, partitionsDefName, type}) => {
+      partitionKeys?.forEach((key) => {
+        rows.push({
+          key,
+          def: partitionsDefName,
+          type,
+        });
+      });
+    });
+    return rows;
+  }, [requests]);
+
+  if (!rows.length) {
+    return null;
+  }
+
+  return (
+    <Box flex={{direction: 'column', gap: 12, grow: 1}} margin={{top: 24}}>
+      <Box flex={{direction: 'row', alignItems: 'center', gap: 8}}>
+        <Icon name="partition" />
+        <Subheading>Dynamic Partition Requests</Subheading>
+      </Box>
+      <Table style={{borderRight: `1px solid ${Colors.KeylineGray}`}}>
+        <thead>
+          <tr>
+            <th>Partition</th>
```

Comment:
What do you think about changing the order of the columns to first group by partitions def name, and then partition?

I think it might be a bit easier to parse if multiple partitions defs requests are returned in one sensor tick

Reply:
We could also make the "partition" column a "partitions" column that contains a list of partitions that have been requested, so that each row matches a request object.

I don't feel strongly about this though--up to you.

Reply:
@braunjj any thoughts re the above?